errors/middleware-upgrade-guide.md

```diff
@@ -34,13 +36,20 @@ If you're using Next.js on Vercel, your existing deploys using Middleware will c
 
 ### Explanation
 
-Previously, you could create a `_middleware.ts` file under the `pages` directory at any level. Middleware execution was based on the file path where it was created. Beta customers found this route matching confusing. For example:
+Previously, you could create a `_middleware.ts` file under the `pages` directory at any level. Middleware execution was based on the file path where it was created.
 
-- Middleware in `pages/dashboard/_middleware.ts`
-- Middleware in `pages/dashboard/users/_middleware.ts`
-- A request to `/dashboard/users/*` **would match both.**
+Based on customer feedback, we have replaced this API with a single root Middleware, which provides the following improvements:
 
-Based on customer feedback, we have replaced this API with a single root Middleware.
+- **Faster execution with lower latency**  
+  With nested middleware, a single request could invoke multiple middleware functions. A single middleware means a single function execution, which is more efficient.
```

Comment:
Can we put these like:

- **Thing:** description

Reply:
Done. 👍